lib/utils/config.go

```diff
@@ -17,24 +17,32 @@ limitations under the License.
 package utils
 
 import (
+	"fmt"
 	"os"
+	"path/filepath"
 	"strings"
 
 	"github.com/gravitational/trace"
 )
 
-// ReadToken is a utility function to read the token
-// from the disk if it looks like a path,
-// otherwise, treat it as a value
-func ReadToken(token string) (string, error) {
-	if !strings.HasPrefix(token, "/") {
-		return token, nil
+// TryReadValueAsFile is a utility function to read a value
+// from the disk if it looks like an absolute path,
+// otherwise, treat it as a value.
+// It only support absolute paths to avoid ambiguity in interpretation of the value
+func TryReadValueAsFile(value string) (string, error) {
+	if !filepath.IsAbs(value) {
+		return value, nil
 	}
-	// treat it as a file
-	out, err := os.ReadFile(token)
+	// treat it as an absolute filepath
+	contents, err := os.ReadFile(value)
 	if err != nil {
 		return "", trace.ConvertSystemError(err)
 	}
 	// trim newlines as tokens in files tend to have newlines
-	return strings.TrimSpace(string(out)), nil
+	out := strings.TrimSpace(string(contents))
+
+	if out == "" {
+		fmt.Fprintf(os.Stderr, "WARNING: empty config value file: %v\n", value)
```

Comment:
I would ditch the warning. Either the empty value is allowed and the warning is false, or the empty value is not allowed and the caller should escalate to a proper error.

Reply:
If teleport.auth_token or auth_service.tokens have a path to an empty file it's currently allowed, but may be surprising to a user having problems getting their node to join a cluster. The idea is to make non-obvious config mistakes obvious to a user, so I'd like to print some kind of message telling them/warning them about it.

Assuming we do want to point out a non-obvious mistake like this, I can't see a case where the caller wouldn't want them to be at least notified - only cases where we decide to escalate to an error as well.

Reply:
Got it.

Are you avoiding `log.Warnf` here because we think it might not be properly initialized? If yes please add a comment, otherwise let's move to `log.Warnf`.

Reply:
I actually wasn't sure how to handle it. I searched our code and found we use both fmt and logging to print warnings. More locally though, we do use log in the utils package, so I can do that